src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Interop/DotNetInvoker.ts

```diff
@@ -0,0 +1,103 @@
+﻿import { MethodOptions, DotnetMethodArgumentsList } from './MethodOptions';
+import { platform } from '../Environment';
+import { System_String } from '../Platform/Platform';
+import { getRegisteredFunction } from './RegisteredFunction';
+
+export function invokeDotNetMethod<T>(methodOptions: MethodOptions, ...args: any[]): (T | null) {
+    const method = platform.findMethod(
+        "Microsoft.AspNetCore.Blazor.Browser",
+        "Microsoft.AspNetCore.Blazor.Browser.Interop",
+        "JavaScriptInvoke",
+        "InvokeDotnetMethod");
+
+    const packedArgs = packArguments(args);
+
+    const serializedOptions = platform.toDotNetString(JSON.stringify(methodOptions));
+    const serializedArgs = platform.toDotNetString(JSON.stringify(packedArgs));
+    const serializedResult = platform.callMethod(method, null, [serializedOptions, serializedArgs]);
+
+    if (serializedResult !== null && serializedResult !== undefined && (serializedResult as any) !== 0) {
```

Comment:
Just for my understanding, is it actually possible for the `serializedResult` to be `null` or `undefined` or zero? Are you just being defensive or do they all represent different scenarios? If so, could you add a comment indicating what each one encodes, or if not, could we remove the clauses that check for outcomes that won't occur (or if you prefer, `throw` for outcomes that shouldn't be possible)?

Sorry to add work. I just want to be sure that as we continue to maintain this we know what scenarios exist.